src/initializers/diamorphosis.ts

```diff
@@ -2,6 +2,40 @@ import * as diamorphosis from 'diamorphosis';
 import { OrkaOptions } from '../typings/orka';
 import { defaultTo, isBoolean } from 'lodash';
 
+export const producerConfigOverrides = config => {
+  const {
+    KAFKA_PRODUCER_BROKERS,
+    KAFKA_PRODUCER_CERTIFICATES_KEY,
+    KAFKA_PRODUCER_CERTIFICATES_CERT,
+    KAFKA_PRODUCER_CERTIFICATES_CA,
+    KAFKA_PRODUCER_SASL_USERNAME,
+    KAFKA_PRODUCER_SASL_PASSWORD
+  } = process.env;
```

Comment:
🔧 Reading from process.env directly could cause some conflicts with diamorphosis. Eg someone could overwrite some values in config.js and be removed.

Reply:
I chose not to go with config because in that case all the apps would need to change their config files as well. Not only bump orka. Do you want me to include something like:
```
certificates: {
      key: config.kafka.producer.certificates.key || KAFKA_PRODUCER_CERTIFICATES_KEY,
}
```?

Reply:
You could reuse the pattern we have with honeybadger :
```javascript
config.honeybadger = {
    apiKey: '',
    ...config.honeybadger
  };

if(config.kafka){
  config.kafka.producer ={
  brokers: [],
  certificates: {
      key:'',
      cert: '',
      ca: '',
     ...config.kafka?.producer?.certificates
    },
    sasl: {
      username: '',
      password: '',
     ...config.kafka?.producer?.sasl
    },
   ...config.kafka.producer
  }
}
```

before diamorphosis is called. And let diamorphosis do the replacements with the env.
After diamoprhosis is called you would then have to give defaults to these values eg:

```js
if(config.kafka){
  config.kafka.producer ={
  brokers: config.kafka.producer.brokers || config.kafka.brokers,
  certificates: {
      key: config.kafka.producer.certificates.key || config.kafka.certificates.key,
    },
   etc
  }
}
```
Possibly a defaultsDeep method from lodash could also help.
Wdyt?

Reply:
I am ok with the first part, but  I can't do the 2nd part of your comment because imagine the following situation:
```
config: {
  kafka: {
    brokers: ['a', 'b']
    certificates: { key: '1', ca: '2', cert: '3'},
    producer: {
      brokers: ['c', 'd'],
      sasl: { username: 'asd', password: 'cvb'}
    }
  }
}
```
then the consumer will end up with the following config:
```
producer: {
      brokers: ['c', 'd'],
      certificates: { key: '1', ca: '2', cert: '3'},
      sasl: { username: 'asd', password: 'cvb'}
}
```
and it will use the `certificates` mechanism instead of `sasl`